src/components/AnswerForm/index.js

```diff
@@ -1,61 +1,162 @@
-import React, { useState } from 'react';
+import React, { useState, useEffect } from 'react';
+import { useForm } from 'react-hook-form';
 import { useParams } from 'react-router-dom';
 
-import { createAnswer } from '@api';
+import { createAnswer, getProductById } from '@api';
 
-const AnswerForm = ({ question_id }) => {
-  const [formData, setFormData] = useState({});
+import Button from '@components/Button';
+import { StyledForm, StyledFormRow, StyledLabel } from './styles';
 
-  const doHandleChange = (e) => {
-    setFormData({
-      ...formData,
-      [e.target.name]: e.target.value,
-    });
-  };
+const AnswerFrom = ({ question, setIsShowing }) => {
+  const {
+    register,
+    handleSubmit,
+    reset,
+    formState: { errors },
+  } = useForm();
+  const { id } = useParams();
+  const [productName, setProductName] = useState('');
+
+  useEffect(async () => {
+    async function fetchData() {
+      const response = await getProductById(id);
```

Comment:
`getProductById` should just return a `product`. In the definition for `getProductId` try to put error handling in there, and on success, just return the product. that way your implementation here can just read as 
```node.js
const { name: productName } = await getProductById(id)
setProductName(productName)
```